src/redux-middleware/fetchBeforeMove.ts

```diff
@@ -0,0 +1,36 @@
+import { ThunkMiddleware } from 'redux-thunk'
+import { pullDescendants } from '../action-creators/pull'
+import { isPending, pathExists } from '../selectors'
+import { hashContext, head, parentOf, pathToContext } from '../util'
+import { State } from '../util/initialState'
+
+/**
+ * Middleware that fetches any pending descendants before a thought-move operation.
+ */
+const fetchBeforeMove: ThunkMiddleware<State> = ({ getState, dispatch }) => {
+
+  return next => async action => {
+
+    // reset internal state variables when clear action is dispatched
+    if (action.type === 'existingThoughtMove') {
+      const state = getState()
+
+      const oldContext = pathToContext(action.oldPath)
+
+      const newContext = pathToContext(action.newPath)
```

Comment:
You should simplify the Paths before converting to Contexts. Otherwise it will break context views. All Contexts must be simple, we just don't have that built into the `pathToContext` type yet.